File: backend/internal/model/squad.go
package model

import (
	"encoding/json"
	"time"
)

// Squad represents a multi-agent collaboration team
type Squad struct {
	BaseModel
	WorkspaceID    uint64         `gorm:"not null;index" json:"workspace_id"`
	ProjectID      *uint64        `gorm:"index" json:"project_id,omitempty"`
	Name           string         `gorm:"size:100;not null" json:"name"`
	Description    string         `gorm:"type:text" json:"description"`
	LeaderAgentID  *uint64        `gorm:"index" json:"leader_agent_id,omitempty"`
	Status         string         `gorm:"size:20;not null;default:active" json:"status"`
	Config         json.RawMessage `gorm:"type:jsonb;default:'{}'" json:"config"`
	Goal           string         `gorm:"type:text" json:"goal"`
	Members        []SquadMember  `json:"members"`
	CreatedAt      time.Time      `json:"created_at"`
	UpdatedAt      time.Time      `json:"updated_at"`
	DeletedAt      *time.Time     `gorm:"index" json:"deleted_at,omitempty"`
}

// SquadMember represents a member agent in a squad
type SquadMember struct {
	BaseModel
	SquadID     uint64     `gorm:"not null;index" json:"squad_id"`
	AgentID     uint64     `gorm:"not null;index" json:"agent_id"`
	Role        string     `gorm:"size:50;not null" json:"role"` // leader, member, observer
	AgentConfigID uint64   `gorm:"index" json:"agent_config_id"`
	Status      string     `gorm:"size:20;not null;default:active" json:"status"`
	AssignedAt  time.Time  `json:"assigned_at"`
	RemovedAt   *time.Time `json:"removed_at,omitempty"`
}

// SquadTask represents a task assigned to a squad member
type SquadTask struct {
	BaseModel
	SquadID         uint64     `gorm:"not null;index" json:"squad_id"`
	AgentTaskID     uint64     `gorm:"not null;index" json:"agent_task_id"`
	MemberID        uint64     `gorm:"not null;index" json:"member_id"`
	Status          string     `gorm:"size:20;not null;default:pending" json:"status"`
	Priority        string     `gorm:"size:20" json:"priority"`
	Progress        int        `gorm:"default:0" json:"progress"`
	TaskDescription string     `gorm:"type:text" json:"task_description"` // What the member was asked to do (decomposed subtask)
	Feedback        string     `gorm:"type:text" json:"feedback"`
	StartedAt       *time.Time `json:"started_at,omitempty"`
	CompletedAt     *time.Time `json:"completed_at,omitempty"`
}

// SquadExecution represents a squad's execution session
type SquadExecution struct {
	BaseModel
	SquadID       uint64         `gorm:"not null;index" json:"squad_id"`
	Status        string         `gorm:"size:20;not null;default:pending" json:"status"`
	Goal          string         `gorm:"type:text" json:"goal"`
	InputData     json.RawMessage `gorm:"type:jsonb;default:'{}'" json:"input_data"`
	OutputData    json.RawMessage `gorm:"type:jsonb" json:"output_data"`
	Logs          json.RawMessage `gorm:"type:jsonb;default:'[]'" json:"logs"`
	StartedAt     *time.Time     `json:"started_at,omitempty"`
	CompletedAt   *time.Time     `json:"completed_at,omitempty"`
	FailedAt      *time.Time     `json:"failed_at,omitempty"`
	ErrorInfo     string         `json:"error_info"`
}

File: frontend/src/api/squad.ts
/**
 * Squad API — 多Agent协作团队 API 模块
 */
import api from './index'

export interface Squad {
  id: number
  workspace_id: number
  project_id?: number
  name: string
  description: string
  leader_agent_id?: number
  status: string
  goal: string
  members: SquadMember[]
  execution_count?: number
  created_at: string
  updated_at: string
}

export interface SquadMember {
  id: number
  squad_id: number
  agent_id: number
  role: string
  agent_config_id: number
  status: string
  assigned_at: string
  removed_at?: string
}

export interface SquadExecution {
  id: number
  squad_id: number
  status: string
  goal: string
  input_data?: any
  output_data?: any
  logs?: any[]
  started_at?: string
  completed_at?: string
  failed_at?: string
  error_info?: string
  created_at: string
}

export interface SquadCreateRequest {
  name: string
  description?: string
  leader_agent_id?: number
  project_id?: number
  goal?: string
  config?: Record<string, any>
  members?: { agent_id: number; role: string; agent_config_id: number }[]
}

export interface SquadUpdateRequest {
  name?: string
  description?: string
  leader_agent_id?: number
  goal?: string
  config?: Record<string, any>
}

export interface SquadMemberAddRequest {
  agent_id: number
  role: string
  agent_config_id: number
}

export interface SquadExecutionStartRequest {
  goal: string
  input_data?: Record<string, any>
}

export async function getSquads(workspaceId: number, projectId?: number): Promise<Squad[]> {
  const params: Record<string, any> = {}
  if (projectId) params.project_id = projectId
  const response = await api.get(`/workspaces/${workspaceId}/squads`, { params })
  return response.data
}

export async function listSquads(workspaceId: number): Promise<Squad[]> {
  const response = await api.get(`/workspaces/${workspaceId}/squads`)
  return response.data
}

export async function createSquad(workspaceId: number, data: SquadCreateRequest): Promise<Squad> {
  const response = await api.post(`/workspaces/${workspaceId}/squads`, data)
  return response.data
}

export async function getSquad(workspaceId: number, squadId: number): Promise<Squad> {
  const response = await api.get(`/workspaces/${workspaceId}/squads/${squadId}`)
  return response.data
}

export async function updateSquad(workspaceId: number, squadId: number, data: SquadUpdateRequest): Promise<Squad> {
  const response = await api.put(`/workspaces/${workspaceId}/squads/${squadId}`, data)
  return response.data
}

export async function deleteSquad(workspaceId: number, squadId: number): Promise<void> {
  await api.delete(`/workspaces/${workspaceId}/squads/${squadId}`)
}

export async function addMember(workspaceId: number, squadId: number, data: SquadMemberAddRequest): Promise<SquadMember> {
  const response = await api.post(`/workspaces/${workspaceId}/squads/${squadId}/members`, data)
  return response.data
}

export async function removeMember(workspaceId: number, squadId: number, memberId: number): Promise<void> {
  await api.delete(`/workspaces/${workspaceId}/squads/${squadId}/members/${memberId}`)
}

export async function startExecution(workspaceId: number, squadId: number, data: SquadExecutionStartRequest): Promise<SquadExecution> {
  const response = await api.post(`/workspaces/${workspaceId}/squads/${squadId}/executions`, data)
  return response.data
}

export async function getExecutions(workspaceId: number, squadId: number): Promise<SquadExecution[]> {
  const response = await api.get(`/workspaces/${workspaceId}/squads/${squadId}/executions`)
  return response.data
}

export async function getExecution(workspaceId: number, squadId: number, executionId: number): Promise<SquadExecution> {
  const response = await api.get(`/workspaces/${workspaceId}/squads/${squadId}/executions/${executionId}`)
  return response.data
}

export async function cancelExecution(workspaceId: number, squadId: number, executionId: number): Promise<void> {
  await api.delete(`/workspaces/${workspaceId}/squads/${squadId}/executions/${executionId}`)
}

export default {
  getSquads,
  listSquads,
  createSquad,
  getSquad,
  updateSquad,
  deleteSquad,
  addMember,
  removeMember,
  startExecution,
  getExecutions,
  getExecution,
  cancelExecution,
}
